src/nvim/api/command.c

```diff
@@ -121,9 +147,63 @@ Dict(cmd) nvim_parse_cmd(String str, Dict(empty) *opts, Arena *arena, Error *err
   Array args = ARRAY_DICT_INIT;
   size_t length = strlen(ea.arg);
 
-  // For nargs = 1 or '?', pass the entire argument list as a single argument,
-  // otherwise split arguments by whitespace.
-  if (ea.argt & EX_NOSPC) {
+  // Check if this is a mapping command that needs special handling
+  // like mapping commands need special argument parsing to preserve whitespace in RHS:
+  // "map a b  c" => { args=["a", "b  c"], ... }
+  bool is_mapping_cmd = false;
+  switch (ea.cmdidx) {
+  case CMD_map:
```

Comment:
this also should be in a function. presumably this handling is needed elsewhere, otherwise how does vim handle :map commands?

Reply:
Then you can define a function in ex_docmd.c that checks the value `cmdnames[ea.cmdidx].cmd_func`.